### DISCUSSION TITLE
Node not found - create it?

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10154] [General Discussions]
If you are working on a script and it has a reference to a node that does not exist, maybe Godot could offer to make this in it's error text?

Example:

`@onready var camera = $Head/Camera`

Say for example this camera node does not exist in the scene tree. This would throw a script error, you could then click on the script error line and a popup will appear offering to make that node that is missing.

Thoughts? or is this too rare?

**Comment by AThousandShips** [comment]
The error is far more likely to happen because you have a typo or you launched the wrong scene, I don't expect many users to ever have this situation because they forgot to create the node in the first place, so new users would create the node incorrectly assuming it's the way to go when in reality they typed the name wrong or did something else wrong (in my experience inexperienced users are generally likely to just press "yes" on any scary popup they see)

I've never hat the situation where the issue is that I forgot to create the node, only that I typed it wrong or changed the path etc.

**Comment by Mickeon** [comment]
I can see the utility in this but there's no such system for the errors sent to handle it "nicely". This situation would have to be hard-coded into the error tab itself, and doing so would be _very_ unreliable. Dedicating a whole popup for this niche case may also not be worth it.
